sympy/vector/basisdependent.py

```diff
@@ -179,6 +180,12 @@ def doit(self, **hints):
                            for x in self.components]
         return self._add_func(*doit_components)
 
+    def limit(self, *args, **kwargs):
+        """Calls limit() on each component"""
+        limit_components = [limit(v, *args, **kwargs) * k for
+                           k, v in self.components.items()]
+        return self._add_func(*limit_components)
```

Comment:
Does this work with the top-level `limit` function or `Limit(v).doit()`?

Reply:
Nope. I also thought if it might be good to also have it working with the top level limit functions (for example by implementing an `_eval_limit` function), but then I saw that e.g. limits in matrices ([here](https://github.com/sympy/sympy/blob/75a7787f2eb3176b5ea17503ee8fff14f3de9102/sympy/matrices/matrixbase.py#L3509)) also only provide the local limit function. 

Should we think about a "bigger" refactoring to make all local limit functionalities available using the top level functions?

Reply:
I guess there should be something like `_eval_limit`.

Maybe @anutosh491 has some ideas what is a good way to do this.

Reply:
HI, 

I shall go through this soon. I am just guessing we might need more tests (just basic ones to start with) if we add some new functionality.

Reply:
Would it be ok to merge the proposed `limit` class function into master for now and separate refactoring of the top-level `limit` function to a separate issue? ;)

Reply:
Some questions / suggestions 

1)
> Would it be ok to merge the proposed limit class function into master for now and separate refactoring of the top-level limit function to a separate issue? ;)

If we are doing this, I need a dedicated issue to track the refactoring from the topic .. otherwise this would be lost. As pointed out above `Limit(f(x) ...).doit()` should be technically responsible for all limits.

2) Is limit of vectors ... as easy as splitting about components and applying limits separately ?
3) Is there no restriction on the type of arg ? (maybe something like what the diff func does)
```
        for x in args:
            if isinstance(x, BasisDependent):
                raise TypeError("Invalid arg for differentiation")        
```
4) Is there no corner case or anything to test this out. The tests added look basic and not sure does justice to the method.
5) So we split about components and then call the limit from the top and return back whatever error we come across from the limits module. Should we catch the error if any ... and return something custom in respect to how limits work for vector (maybe a different message or why stuff failed) ? 

These are some basic question and I haven't worked on this module before.

Reply:
Exactly I was curious about things like these. Adding such a limit function and a basic test look too good to be true and doesn't guarantee everything would work.